File: Leetcode/Easy/169 - Majority Element.py
class Solution:
    def majorityElement(self, nums: [int]) -> int:
        count = {}
        for x in range(len(nums)):
            if nums[x] in count.keys():
                count[nums[x]] = count[nums[x]] + 1
                continue
            count[nums[x]] = 1
        for num in count.keys():
            if count[num] == max(count.values()):
                return num

solution = Solution()
print(solution.majorityElement([1, 2, 3, 1, 5, 1]))
print(solution.majorityElement([1, 2, 3, 1, 5, 1, 1, 1, 1, 1, 1]))
print(solution.majorityElement([2, 2, 2]))

File: Leetcode/Easy/JavaScript/169 - Majority Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var elementCount = [];
    for(let i=0; i<nums.length; i++){
        if(elementCount[nums[i]] == undefined){
            elementCount[nums[i]] = 1;
        }
        else{
            elementCount[nums[i]] += 1;
        }
        if(elementCount[nums[i]] > nums.length / 2){
            return nums[i];
        }
    }
};

console.log(majorityElement([1,2,2,1,2,2,2,1])); // 2
console.log(majorityElement([1])); // 1